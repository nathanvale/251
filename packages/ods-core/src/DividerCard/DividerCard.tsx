import React, { Children } from "react";
import styled, { css } from "styled-components";
import {
  ResponsiveSpace,
  BreakpointVariants,
  CardPaddingVariants,
  Theme,
  RequiredNoBaseProps,
} from "@origin-digital/ods-types";
import { mapToStyledSystem } from "@origin-digital/ods-helpers";
import { Box, BoxProps } from "../Box";
import { defaultCardProps, cardPaddingForVariant, CardProps } from "../Card";
import { StackChild, StackChildProps } from "../Stack";
import { stackDefaultProps } from "../Stack/Stack";

export interface DividerCardProps extends CardProps {
  space?: ResponsiveSpace;
}
const dividerCardDefaultProps: RequiredNoBaseProps<DividerCardProps> = {
  "data-id": "divider-card",
  padding: defaultCardProps.padding,
  backgroundColor: defaultCardProps.backgroundColor,
  space: stackDefaultProps.space,
};

interface DividerCardChildProps {
  cardPadding: CardPaddingVariants;
}

const childPadding = ({
  cardPadding,
  screenSize,
  theme,
}: DividerCardChildProps & {
  screenSize: BreakpointVariants;
  theme: Theme;
}) => {
  const paddingX = cardPaddingForVariant[cardPadding].paddingX[screenSize];

  return css`
    ${paddingX
      ? `
          padding-left: ${theme.space[paddingX]}px;
          padding-right: ${theme.space[paddingX]}px;

          &:not(:empty) ~ div:not(:empty):before,
          &:nth-child(n + 2):before {
            margin-left: -${theme.space[paddingX]}px;
            margin-right: -${theme.space[paddingX]}px;
          }
        `
      : ``}
  `;
};

const DividerCardChild = styled<
  DividerCardChildProps & StackChildProps & BoxProps
>(({ cardPadding, ...stackChildProps }) => <StackChild {...stackChildProps} />)`
  ${({ cardPadding, theme }) => {
    const screens: BreakpointVariants[] = ["sm", "md", "lg", "xl"];
    return css`
      ${childPadding({ theme, cardPadding, screenSize: "xs" })}

      ${screens.map(
        (screenSize) => css`
          @media (min-width: ${theme.breakpoints[screenSize]}) {
            ${childPadding({ theme, cardPadding, screenSize })}
          }
        `
      )}
    `;
  }}
`;

export const DividerCard = ({
  backgroundColor = dividerCardDefaultProps.backgroundColor,
  children,
  padding = dividerCardDefaultProps.padding,
  space,
  "data-id": dataId,
}: DividerCardProps) => {
  const stackItems = Children.toArray(children);
  const { paddingY } = cardPaddingForVariant[padding];

  return (
    <Box
      data-id={dataId}
      display="block"
      width="full"
      backgroundColor={backgroundColor}
      paddingY={paddingY}
    >
      {stackItems.flatMap((child, index) => {
        return (
          <DividerCardChild
            divider
            display="block"
            key={index}
            data-id="divider-card-child"
            space={mapToStyledSystem(space)}
            cardPadding={padding}
          >
            {child}
          </DividerCardChild>
        );
      })}
    </Box>
  );
};

DividerCard.defaultProps = dividerCardDefaultProps;
DividerCard.displayName = "DividerCard";
