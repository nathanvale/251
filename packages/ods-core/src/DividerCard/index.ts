export { DividerCard } from "./DividerCard";
export type DividerCardProps = import("./DividerCard").DividerCardProps;
