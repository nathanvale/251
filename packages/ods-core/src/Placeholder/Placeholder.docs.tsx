import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Placeholder, PlaceholderProps } from "./Placeholder";

export const docs: ComponentDocs<PlaceholderProps> = {
  category: "Layout",
  componentName: "Placeholder",
  description:
    "This component is helpful for laying out placeholders for future content.",
  propDescriptions: {},
  migrationGuide: false,
  examples: {
    default: {
      Code: () => <Placeholder />,
    },
    additional: [
      {
        label: "Setting the Label",
        Code: () => <Placeholder label="Placeholder for content" />,
      },
      {
        label: "Setting the Shape",
        Code: () => <Placeholder shape="round" width={120} height={120} />,
      },
      {
        label: "Setting the Width",
        Code: () => <Placeholder width={120} />,
      },
      {
        label: "Setting the Height",
        Code: () => <Placeholder height={400} />,
      },
      {
        label: "Responsive Height Across Mobile and Desktop",
        description:
          "Width can be controlled responsively by passing in an array of mobile and desktop pixel sizes as values. Open in Playroom to see the responsive behaviour of this example.",
        Code: () => <Placeholder height={[100, 500]} />,
      },
      {
        label: "Responsive Height Across All Five Breakpoints",
        description:
          "For more granular control across all 5 breakpoints, ‘xs’, ‘sm’,  ‘md’, ‘lg’ and ‘xl’, width can be controlled responsively by passing in an object with breakpoints as keys and pixel sizes as values. Open in Playroom to see the responsive behaviour of this example.",
        Code: () => (
          <Placeholder
            height={{
              xs: 100,
              sm: 200,
              md: 300,
              lg: 400,
              xl: 500,
            }}
          />
        ),
      },
    ],
  },
  snippets: [],
};
