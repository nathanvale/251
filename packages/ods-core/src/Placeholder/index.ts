export { Placeholder } from "./Placeholder";
export type PlaceholderProps = import("./Placeholder").PlaceholderProps;
