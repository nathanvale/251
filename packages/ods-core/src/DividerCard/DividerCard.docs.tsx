/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Box, TextLink, Text } from "@origin-digital/ods-core";
import { Placeholder } from "../Placeholder/Placeholder";
import { DividerCardProps, DividerCard as ODSDivider } from "./DividerCard";

const DividerCard = (props: DividerCardProps) => (
  <Box padding="xxlarge" backgroundColor="grey200">
    <ODSDivider {...props} />
  </Box>
);

export const docs: ComponentDocs<DividerCardProps> = {
  category: "Layout",
  componentName: "DividerCard",
  description: (
    <Text>
      DividerCard works the same as{" "}
      <TextLink href="#/components/Card">Card</TextLink> except it adds a
      divider between each of it's children the same way{" "}
      <TextLink href="#/components/Stack">Stack</TextLink> does.
    </Text>
  ),
  propDescriptions: {
    padding: "The size of the responsive paddingX and paddingY of the card.",
    space:
      "The T-shirt sized vertical space that DividerCard will add between its children.",
  },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => (
        <DividerCard>
          <Placeholder />
          <Placeholder />
        </DividerCard>
      ),
    },
    additional: [
      {
        label: "Medium padding and medium spacing",
        Code: () => (
          <DividerCard padding="medium" space="medium">
            <Placeholder />
            <Placeholder />
          </DividerCard>
        ),
      },
      {
        label: "Large padding and large spacing",
        Code: () => (
          <DividerCard padding="large" space="large">
            <Placeholder />
            <Placeholder />
          </DividerCard>
        ),
      },
    ],
  },
  snippets: [
    {
      label: "Default",
      Code: () => (
        <DividerCard>
          <Placeholder />
          <Placeholder />
        </DividerCard>
      ),
    },
    {
      label: "Medium padding and medium spacing",
      Code: () => (
        <DividerCard padding="medium" space="medium">
          <Placeholder />
          <Placeholder />
        </DividerCard>
      ),
    },
    {
      label: "Large padding and large spacing",
      Code: () => (
        <DividerCard padding="large" space="large">
          <Placeholder />
          <Placeholder />
        </DividerCard>
      ),
    },
  ],
};
