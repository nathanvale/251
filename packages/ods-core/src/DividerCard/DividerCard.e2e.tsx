import React from "react";
import { E2ETests } from "@origin-digital/ods-types";
import { Placeholder } from "../Placeholder";
import { DividerCard } from "./DividerCard";

const RenderNull = () => null;

export const tests: E2ETests = [
  {
    label: "Default card",
    Code: () => (
      <DividerCard>
        <Placeholder />
        <Placeholder />
        <Placeholder />
      </DividerCard>
    ),
    responsive: true,
  },
  {
    label: "Medium padding and medium space",
    Code: () => (
      <DividerCard padding="medium" space="medium">
        <Placeholder />
        <Placeholder />
        <Placeholder />
      </DividerCard>
    ),
    responsive: true,
  },
  {
    label: "Large padding and large space",
    Code: () => (
      <DividerCard padding="large" space="large">
        <Placeholder />
        <Placeholder />
        <Placeholder />
      </DividerCard>
    ),
    responsive: true,
  },
  {
    label: "Rendered with null children",
    Code: () => (
      <DividerCard space="xxlarge">
        <RenderNull />
        <Placeholder />
        <RenderNull />
        <Placeholder />
        <RenderNull />
        <Placeholder />
        <RenderNull />
      </DividerCard>
    ),
  },
];
