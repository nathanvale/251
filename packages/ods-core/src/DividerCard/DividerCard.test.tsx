import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { Placeholder } from "../Placeholder";
import { DividerCard, DividerCardProps } from "./DividerCard";

const RenderNull = () => null;

const renderDividerCard = (props: DividerCardProps = {}) =>
  render(
    <DividerCard {...props}>
      <Placeholder />
      <Placeholder />
    </DividerCard>
  );

test("it renders a divider card", () => {
  const { container } = renderDividerCard();
  expect(container).toMatchSnapshot();
});

test("it renders a divider card with custom data-id", () => {
  const { container } = renderDividerCard({ "data-id": "myId" });
  expect(container.firstChild).toHaveAttribute("data-id", "myId");
});

test("it renders a divider card with medium padding and medium spacing", () => {
  const { container } = renderDividerCard({
    padding: "medium",
    space: "medium",
  });
  expect(container).toMatchSnapshot();
});

test("it renders a divider card with large padding", () => {
  const { container } = renderDividerCard({ padding: "large", space: "large" });
  expect(container).toMatchSnapshot();
});

test("it renders a divider card with a selected background color", () => {
  const { container } = renderDividerCard({
    padding: "large",
    space: "large",
    backgroundColor: "grey50",
  });
  expect(container).toMatchSnapshot();
});

test("It can distribute space with null children", () => {
  const { container } = render(
    <DividerCard space="xxlarge">
      <RenderNull />
      <Placeholder />
      <RenderNull />
      <Placeholder />
      <RenderNull />
      <Placeholder />
      <RenderNull />
    </DividerCard>
  );
  expect(container).toMatchSnapshot();
});
