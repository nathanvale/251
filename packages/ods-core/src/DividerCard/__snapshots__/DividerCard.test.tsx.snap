// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It can distribute space with null children 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: white;
  display: block;
  padding-top: 24px;
  padding-bottom: 24px;
  width: 100%;
}

.c3 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
}

.c5 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
}

.c2:empty {
  display: none;
}

.c2:empty + div:before {
  display: none;
}

.c2:not(:empty) ~ div:not(:empty):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  height: 1px;
  box-shadow: inset 0 0 0 1px #e3e3e3;
  margin-bottom: 47px;
  margin-top: 48px;
}

.c2:nth-child(n + 2):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  height: 1px;
  box-shadow: inset 0 0 0 1px #e3e3e3;
  margin-bottom: 47px;
  margin-top: 48px;
}

.c1 {
  padding-left: 16px;
  padding-right: 16px;
}

.c1:not(:empty) ~ div:not(:empty):before,
.c1:nth-child(n + 2):before {
  margin-left: -16px;
  margin-right: -16px;
}

.c4 {
  width: 100%;
  min-width: 10px;
  height: 120px;
  border: 2px solid rgba(51,51,51,0.3);
}

@media screen and (min-width:992px) {
  .c0 {
    padding-top: 32px;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    padding-bottom: 32px;
  }
}

@media (min-width:576px) {
  .c1 {
    padding-left: 32px;
    padding-right: 32px;
  }

  .c1:not(:empty) ~ div:not(:empty):before,
  .c1:nth-child(n + 2):before {
    margin-left: -32px;
    margin-right: -32px;
  }
}

<div>
  <div
    class="c0"
    data-id="divider-card"
    display="block"
    width="full"
  >
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    />
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    >
      <div
        class="c4 c5"
        data-id="placeholder"
        display="flex"
        height="120"
        overflow="hidden"
        shape="rectangle"
        width="100%"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    />
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    >
      <div
        class="c4 c5"
        data-id="placeholder"
        display="flex"
        height="120"
        overflow="hidden"
        shape="rectangle"
        width="100%"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    />
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    >
      <div
        class="c4 c5"
        data-id="placeholder"
        display="flex"
        height="120"
        overflow="hidden"
        shape="rectangle"
        width="100%"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    />
  </div>
</div>
`;

exports[`it renders a divider card 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: white;
  display: block;
  padding-top: 24px;
  padding-bottom: 24px;
  width: 100%;
}

.c3 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
}

.c5 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
}

.c2:empty {
  display: none;
}

.c2:empty + div:before {
  display: none;
}

.c2:not(:empty) ~ div:not(:empty):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  height: 1px;
  box-shadow: inset 0 0 0 1px #e3e3e3;
  margin-bottom: -1px;
  margin-top: 0px;
}

.c2:nth-child(n + 2):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  height: 1px;
  box-shadow: inset 0 0 0 1px #e3e3e3;
  margin-bottom: -1px;
  margin-top: 0px;
}

.c1 {
  padding-left: 16px;
  padding-right: 16px;
}

.c1:not(:empty) ~ div:not(:empty):before,
.c1:nth-child(n + 2):before {
  margin-left: -16px;
  margin-right: -16px;
}

.c4 {
  width: 100%;
  min-width: 10px;
  height: 120px;
  border: 2px solid rgba(51,51,51,0.3);
}

@media screen and (min-width:992px) {
  .c0 {
    padding-top: 32px;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    padding-bottom: 32px;
  }
}

@media (min-width:576px) {
  .c1 {
    padding-left: 32px;
    padding-right: 32px;
  }

  .c1:not(:empty) ~ div:not(:empty):before,
  .c1:nth-child(n + 2):before {
    margin-left: -32px;
    margin-right: -32px;
  }
}

<div>
  <div
    class="c0"
    data-id="divider-card"
    display="block"
    width="full"
  >
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    >
      <div
        class="c4 c5"
        data-id="placeholder"
        display="flex"
        height="120"
        overflow="hidden"
        shape="rectangle"
        width="100%"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    >
      <div
        class="c4 c5"
        data-id="placeholder"
        display="flex"
        height="120"
        overflow="hidden"
        shape="rectangle"
        width="100%"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders a divider card with a selected background color 1`] = `
.c3 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
}

.c5 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
}

.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #f8f8f8;
  display: block;
  padding-top: 24px;
  padding-bottom: 24px;
  width: 100%;
}

.c2:empty {
  display: none;
}

.c2:empty + div:before {
  display: none;
}

.c2:not(:empty) ~ div:not(:empty):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  height: 1px;
  box-shadow: inset 0 0 0 1px #e3e3e3;
  margin-bottom: 23px;
  margin-top: 24px;
}

.c2:nth-child(n + 2):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  height: 1px;
  box-shadow: inset 0 0 0 1px #e3e3e3;
  margin-bottom: 23px;
  margin-top: 24px;
}

.c1 {
  padding-left: 16px;
  padding-right: 16px;
}

.c1:not(:empty) ~ div:not(:empty):before,
.c1:nth-child(n + 2):before {
  margin-left: -16px;
  margin-right: -16px;
}

.c4 {
  width: 100%;
  min-width: 10px;
  height: 120px;
  border: 2px solid rgba(51,51,51,0.3);
}

@media screen and (min-width:576px) {
  .c0 {
    padding-top: 48px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-top: 64px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-bottom: 48px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-bottom: 64px;
  }
}

@media (min-width:576px) {
  .c1 {
    padding-left: 48px;
    padding-right: 48px;
  }

  .c1:not(:empty) ~ div:not(:empty):before,
  .c1:nth-child(n + 2):before {
    margin-left: -48px;
    margin-right: -48px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding-left: 64px;
    padding-right: 64px;
  }

  .c1:not(:empty) ~ div:not(:empty):before,
  .c1:nth-child(n + 2):before {
    margin-left: -64px;
    margin-right: -64px;
  }
}

<div>
  <div
    class="c0"
    data-id="divider-card"
    display="block"
    width="full"
  >
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    >
      <div
        class="c4 c5"
        data-id="placeholder"
        display="flex"
        height="120"
        overflow="hidden"
        shape="rectangle"
        width="100%"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    >
      <div
        class="c4 c5"
        data-id="placeholder"
        display="flex"
        height="120"
        overflow="hidden"
        shape="rectangle"
        width="100%"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders a divider card with large padding 1`] = `
.c3 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
}

.c5 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
}

.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: white;
  display: block;
  padding-top: 24px;
  padding-bottom: 24px;
  width: 100%;
}

.c2:empty {
  display: none;
}

.c2:empty + div:before {
  display: none;
}

.c2:not(:empty) ~ div:not(:empty):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  height: 1px;
  box-shadow: inset 0 0 0 1px #e3e3e3;
  margin-bottom: 23px;
  margin-top: 24px;
}

.c2:nth-child(n + 2):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  height: 1px;
  box-shadow: inset 0 0 0 1px #e3e3e3;
  margin-bottom: 23px;
  margin-top: 24px;
}

.c1 {
  padding-left: 16px;
  padding-right: 16px;
}

.c1:not(:empty) ~ div:not(:empty):before,
.c1:nth-child(n + 2):before {
  margin-left: -16px;
  margin-right: -16px;
}

.c4 {
  width: 100%;
  min-width: 10px;
  height: 120px;
  border: 2px solid rgba(51,51,51,0.3);
}

@media screen and (min-width:576px) {
  .c0 {
    padding-top: 48px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-top: 64px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-bottom: 48px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-bottom: 64px;
  }
}

@media (min-width:576px) {
  .c1 {
    padding-left: 48px;
    padding-right: 48px;
  }

  .c1:not(:empty) ~ div:not(:empty):before,
  .c1:nth-child(n + 2):before {
    margin-left: -48px;
    margin-right: -48px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding-left: 64px;
    padding-right: 64px;
  }

  .c1:not(:empty) ~ div:not(:empty):before,
  .c1:nth-child(n + 2):before {
    margin-left: -64px;
    margin-right: -64px;
  }
}

<div>
  <div
    class="c0"
    data-id="divider-card"
    display="block"
    width="full"
  >
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    >
      <div
        class="c4 c5"
        data-id="placeholder"
        display="flex"
        height="120"
        overflow="hidden"
        shape="rectangle"
        width="100%"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    >
      <div
        class="c4 c5"
        data-id="placeholder"
        display="flex"
        height="120"
        overflow="hidden"
        shape="rectangle"
        width="100%"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders a divider card with medium padding and medium spacing 1`] = `
.c3 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
}

.c5 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
}

.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: white;
  display: block;
  padding-top: 24px;
  padding-bottom: 24px;
  width: 100%;
}

.c2:empty {
  display: none;
}

.c2:empty + div:before {
  display: none;
}

.c2:not(:empty) ~ div:not(:empty):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  height: 1px;
  box-shadow: inset 0 0 0 1px #e3e3e3;
  margin-bottom: 15px;
  margin-top: 16px;
}

.c2:nth-child(n + 2):before {
  content: "";
  display: block;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  position: relative;
  height: 0.1px;
  height: 1px;
  box-shadow: inset 0 0 0 1px #e3e3e3;
  margin-bottom: 15px;
  margin-top: 16px;
}

.c1 {
  padding-left: 16px;
  padding-right: 16px;
}

.c1:not(:empty) ~ div:not(:empty):before,
.c1:nth-child(n + 2):before {
  margin-left: -16px;
  margin-right: -16px;
}

.c4 {
  width: 100%;
  min-width: 10px;
  height: 120px;
  border: 2px solid rgba(51,51,51,0.3);
}

@media screen and (min-width:576px) {
  .c0 {
    padding-top: 48px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    padding-bottom: 48px;
  }
}

@media (min-width:576px) {
  .c1 {
    padding-left: 48px;
    padding-right: 48px;
  }

  .c1:not(:empty) ~ div:not(:empty):before,
  .c1:nth-child(n + 2):before {
    margin-left: -48px;
    margin-right: -48px;
  }
}

<div>
  <div
    class="c0"
    data-id="divider-card"
    display="block"
    width="full"
  >
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    >
      <div
        class="c4 c5"
        data-id="placeholder"
        display="flex"
        height="120"
        overflow="hidden"
        shape="rectangle"
        width="100%"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c1 c2 c3"
      data-id="divider-card-child"
      display="block"
    >
      <div
        class="c4 c5"
        data-id="placeholder"
        display="flex"
        height="120"
        overflow="hidden"
        shape="rectangle"
        width="100%"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
